### ISSUE TITLE
[Question] LibreTime working OS

### ISSUE THREAD

**Issue by Lapotor** [LibreTime/libretime#191]
Hey i have a simple question on everyone.
Which OS do you use with LibreTime and how good does it work on it like
- 3rd party stream
- track upload
- liquidsoap

May i can find a good OS for me ?

**Comment by hairmare** [comment]
I think we should update the [preparing the server](http://libretime.org/manual/preparing-the-server/) docs based on this once we have the right answer. 

> Which OS do you use with LibreTime and how good does it work on it like

Right now I would recommend sticking to Ubuntu Trusty but that might change soon. Personally I use CentOS but I haven't done enough testing with the playout parts to wholesale recommend it and there are still some analyzer issues that need sorting. The docs originally recommend Debian and I'm hoping that will make sense again once we have packages.

> * track upload

As far as I can tell the uploader should work ok in all distros, not all of them support all of analyzers features though (ie. no replaygain on CentOS, sometimes questionable silan values on Debian/Ubuntu).

> * 3rd party stream
> * liquidsoap

3rd party streaming and liquidsoap have issues in how they get started without systemd or upstart. The Ubuntu Trusty upstart configuration takes care of restarting the services should they crash or stop. CentOS does the same by using the systemd configuration from [`installer/systemd`](https://github.com/LibreTime/libretime/tree/master/installer/systemd). Having services get restarted automatically is used as a feature by pypo and it might be broken in some distros.

> Maybe i can find a good OS for me ?

I understand your main use case for the system is LibreTime. For the most parts a particular software should not be the driving force behind deciding which distro to use. Most important is that you pick the one you are most comfortable with. I'll try giving a broader perspective on the choices.

* **Ubuntu** is easy to install and has a large focus on being user friendly, it is based on Debian and comes with optional commercial support from canonical.
* **Debian**  has a strong focus in only including free/libre software, maintained and supported by the Debian community.
* **CentOS** an open version of RedHat Enterprise Linux, here the enterprise aspect means that it aims at being stable and will usually not have modern packages in the core os. 

There are small differences in how Ubuntu and Debian are set up, CentOS is more or less completely different (it uses rpm instead of dpkg/deb). 

If you are new to Linux you should probably consider choosing a distro where you you can find help should anything go wrong. Maybe there is a Linux User Group (LUG) or Hackspace in your listening area, reach out to them. If you are a local community radio station you will probably find that their hackers have been listening to your program all along and are more than willing to help out when it comes to getting your program on air. The key is finding allies so you have someone to be your second set of eyes when you do setup and maintenance work on your production systems.

Sorry for giving a not so simple answer, I'll re-use some of it when I update the docs. I'm very interested in more answers to the original question to help get a better picture for the docs update.

tldr: I use CentOS, all distros currently have issues that need workarounds and/or reporting

**Comment by Robby262** [comment]
Ubuntu Trusty
Track upload works fine
Liquid soap doesn't work on AAC/AAC+

**Comment by Lapotor** [comment]
Yeah i tried Debian 8 but there we have a problem with the 3rd party stream and i think the upload of tracks

>tldr: I use CentOS, all distros currently have issues that need workarounds and/or reporting

Yeah on Ubuntu Xenial, i tried i have the issues #175 and #175 thats why i have a look on other distro where LT work for them.

**Comment by Robby262** [comment]
Ubuntu Trusty works. The LiguidSoap issue isn't really a Trusty Issue, it's a LiquidSoap issue. I would use Trusty. If you don't use AAC, it works fine and with my little patch, that works too.

**Comment by Lapotor** [comment]
which patch do you mean?

**Comment by Robby262** [comment]
Edit the liquidsoap.liq to re-enable the built in aac handler and disable the fdaac line to disable the non working external library. Only applies to trusty. Haremare was not at all happy with this workaround, but it does work fine.

**Comment by hairmare** [comment]
I believe @Robby262 was talking about [aac.txt](https://github.com/LibreTime/libretime/files/949771/acc.txt) that was attached to https://github.com/LibreTime/libretime/issues/185#issuecomment-296446474.

You could also try to install liquidsoap 1.3.0 from opam as described in #192. You'd also need to apply the [patch](https://github.com/LibreTime/libretime/compare/master...radiorabe:feature/liquidsoap-1.3.0.patch) I linked there to LibreTime for that to work.

Robbys change uses the `%aac` encoder that is being phased out by liquidsoap, my solution with liquidsoap 1.3.0 uses `%fdkaac` but you need to patch LibreTime for it to work. I'm not sure which solution sounds better (as in audio quality).

Upstream does seem to be switching to [fdk-aac](https://en.wikipedia.org/wiki/Fraunhofer_FDK_AAC) (the AAC implementation that was open sourced via the Android project). The latter supports the HE-AAC and HE-AACv2 profiles which could be interesting.

Could you explain why you want to use AAC? I think [opus](http://opus-codec.org/) might be a better fit but it really depends on your needs (and liquidsoap 1.1.1 on Debian/Ubuntu most likely won't support opus anyway).

**Comment by Lapotor** [comment]
@Robby262 which version of libretime do you installed ?

**Comment by Lapotor** [comment]
How can i get solve it that when i Stop the stream and after Time x i Start the Stream again liquidsoap die so i Need to Restart IT? Yep i Need IT Tag Why?

**Comment by Robby262** [comment]
I am using LT 3.0.0-alpha.1 installed from the tarball. 

AAC and AAC+ (MPA) are the stated preferred codecs for Apple devices (it is Apple technology).  Approx 30% of my listeners use AAC (96k Stereo), almost none use OGG  (128k Stereo) and the rest use MP3 (128k Stereo).

Most radio directories do not even have an entry for OPUS at all. A recent search in the Xiph directory showed only ONE listed station using OPUS!

The built in AAC Liquidsoap codec works very well, is well supported by the popular Apple apps and we shall use that as long as we use Trusty. It works and it is a very simple script change. The built in AAC+ is really crap. I am told by AlbertS that the fdaac codec in AAC+ mode is excellent.

**Comment by hairmare** [comment]
> How can i get solve it that when i Stop the stream and after Time x i Start the Stream again liquidsoap die so i Need to Restart IT? Yep i Need IT Tag Why?

How are you stopping the stream? On Ubuntu Trusty upstart should take care of ensuring that `airtime-liquidsoap` immediately gets restarted when it "crashes" or is stopped. I'm not sure what happens if you stop icecast.

On systemd distros the [systemd units](https://github.com/LibreTime/libretime/tree/2b30a14744f857657f2e3b7e9c96713929343dd2/installer/systemd) can take care of restarting liquidsoap should it crash. Installing them properly on supported distros is being worked on as a part of #173.

**Comment by Lapotor** [comment]
Will test

**Comment by Robby262** [comment]
A 'Restart Libretime' button in 'Status' (Celery, LS, Playout, Icecast etc) would be very useful.

**Comment by Robbt** [comment]
This issue was moved to http://libretime.trydiscourse.com/t/question-libretime-working-os/29
